From 722eae74b285bae78733f77a05c7e1f3dfd636ec Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@gmail.com>
Date: Fri, 27 Oct 2017 17:24:09 +0200
Subject: [PATCH] make rostest in CMakeLists optional (ros/rosdistro#3010)
 (#75)

* make rostest in CMakeLists optional (ros/rosdistro#3010)
* rostest is now a test_depend instead of a build_depend

Upstream-Status: Accepted [https://github.com/ros-industrial/industrial_core/commit/722eae74b285bae78733f77a05c7e1f3dfd636ec]

This patch has been generated with:
  `git format-patch -1 722eae74b285bae78733f77a05c7e1f3dfd636ec`
in the robot_state_publisher repository.

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@gmail.com>
---
 CMakeLists.txt | 4 +++-
 package.xml    | 3 +--
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c296b2e..76b2af0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -5,7 +5,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -Wextra")
 
 find_package(orocos_kdl REQUIRED)
 find_package(catkin REQUIRED
-  COMPONENTS roscpp rosconsole rostime tf tf2_ros tf2_kdl kdl_parser rostest
+  COMPONENTS roscpp rosconsole rostime tf tf2_ros tf2_kdl kdl_parser
 )
 find_package(Eigen3 REQUIRED)
 
@@ -40,6 +40,8 @@ target_link_libraries(state_publisher ${PROJECT_NAME}_solver ${orocos_kdl_LIBRAR
 
 if (CATKIN_ENABLE_TESTING)
 
+  find_package(rostest REQUIRED)
+
   add_rostest_gtest(test_one_link ${CMAKE_CURRENT_SOURCE_DIR}/test/test_one_link.launch test/test_one_link.cpp)
   target_link_libraries(test_one_link ${catkin_LIBRARIES} ${PROJECT_NAME}_solver)
 
diff --git a/package.xml b/package.xml
index 65de1b4..dee5343 100644
--- a/package.xml
+++ b/package.xml
@@ -30,7 +30,6 @@
   <build_depend version_gte="1.3.0">orocos_kdl</build_depend>
   <build_depend>rosconsole</build_depend>
   <build_depend>roscpp</build_depend>
-  <build_depend>rostest</build_depend>
   <build_depend>rostime</build_depend>
   <build_depend>sensor_msgs</build_depend>
   <build_depend>tf</build_depend>
@@ -49,5 +48,5 @@
   <run_depend>tf2_ros</run_depend>
   <run_depend>tf2_kdl</run_depend>
 
-  <build_depend>rostest</build_depend>
+  <test_depend>rostest</test_depend>
 </package>
-- 
1.9.1

